src/wagtail_vector_index/index/base.py

```diff
@@ -21,6 +27,25 @@ class QueryResponse(Generic[VectorIndexableType]):
     sources: Iterable[VectorIndexableType]
 
 
+@dataclass
+class AsyncQueryResponse(Generic[VectorIndexableType]):
```

Comment:
**Note:** I have to return the whole response object to access it's iterator (response chunks). I’d prefer to avoid this and instead update QueryResponse to return the whole response instead of a string too.

Theres also the case that users might want to access the response object as a whole and for other attributes and methods like `json()`. 

If so maybe we can rename response as "llm_response"?